File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/05_While_Loop_Exercise/04_Walking.js
function walking(input) {
    let goalSteps = 10000;
    let realSteps = 0;

    let i = 0;
    while (true) {
        let currentCommand = input[i];
        i++;

        if (currentCommand == "Going home") {
            let currentStepsGoingHome = Number(input[i]);
            realSteps += currentStepsGoingHome;
            if (realSteps < goalSteps) {
                console.log(`${goalSteps - realSteps} more steps to reach goal.`);
            } else {
                console.log('Goal reached! Good job!');
                console.log(`${realSteps - goalSteps} steps over the goal!`);
            }
            break;
        }

        let currentSteps = Number(currentCommand);
        realSteps += currentSteps;  // realSteps += Number(currentCommand) instead of line 15 & 16

        if (realSteps >= goalSteps) {
            console.log('Goal reached! Good job!');
            console.log(`${realSteps - goalSteps} steps over the goal!`);
            break;
        }
    }

}


walking(["1000",
    "1500",
    "2000",
    "6500"]);

File: Python_Courses/Python_Basics/Python_PB_2023_04/05_While_Loop_Exercise/04_Walking.py
steps = 0

command = input()  # "Going home" OR "55"
while command != "Going home":
    current_steps = int(command)
    steps += current_steps
    if steps >= 10_000:
        break

    command = input()


if command == "Going home":
    current_steps = int(input())
    steps += current_steps

if steps >= 10_000:
    print("Goal reached! Good job!")
    print(f"{steps - 10_000} steps over the goal!")
else:
    print(f"{10_000 - steps} more steps to reach goal.")
